### PR TITLE
Eip8 auth

### PR DESCRIPTION
**Pull request by yglukhov**
Also fixed msgId generation (cc @zah)

### CODE REVIEW

File: ethp2p/rlpx.nim [hunk 1]
```diff
@@ -574,11 +574,20 @@ proc rlpxConnectIncoming*(myKeys: KeyPair, listenPort: Port, address: IpAddress,
   var handshake = newHandshake({Responder})
   handshake.host = myKeys
 
-  var authMsg: array[1024, byte]
+  var authMsg: array[4086, byte]
```
Comment (cheatfate) [inline ethp2p/rlpx.nim]: AuthMessage size can be >= AuthMessageV4Length and <= 2 + 65535.

File: ethp2p/rlpx.nim [hunk 2]
```diff
@@ -574,11 +574,20 @@ proc rlpxConnectIncoming*(myKeys: KeyPair, listenPort: Port, address: IpAddress,
...
+  var ret = handshake.decodeAuthMessage(^authMsg)
+  if ret == AuthStatus.IncompleteError: # Eip8 auth message is likely
+    let expectedLen = expectedAuthMsgLenEip8(authMsg).int + 2
+    if expectedLen > authMsg.len:
+      raise newException(Exception, "Auth message too big: " & $authMsgLen)
+    await s.fullRecvInto(addr authMsg[authMsgLen], expectedLen - authMsgLen)
+    authMsgLen = expectedLen
+    ret = handshake.decodeAuthMessage(^authMsg)
+
+  check ret
```
Comment (cheatfate) [inline ethp2p/rlpx.nim]:
```nim
var authMsg = newSeq[byte](65535 + 2)
var rindex = 0
var rlength = len(authMsg)
while true:
  var res = await recvInto(socket, addr authmsg[rindex], rlength)
  if res == 0:
    ## disconnected
    break
  else:
    let status = decryptAuthMessage(authMsg)
    if status == AuthStatus.IncompleteError:
      rindex += res
      rlength -= res
      continue
    elif status == AuthStatus.Success:
      ## authentication successful
      break
    else:
      ## authentication failed
      break
```

**Comment by cheatfate** [comment]
Usage of `fullRecvInto` is unsafe because it not handle disconnects, and on disconnect it will go into infinite loop.
When `recvInto` returns `0` it means remote peer is disconnected, and this case must be handled properly.

**Comment by yglukhov** [comment]
Also fixed rlpx msg formatting. Depends on https://github.com/status-im/nim-rlp/pull/9. cc @zah

**Comment by yglukhov** [comment]
should be good to go now.

File: ethp2p/rlpx.nim [hunk 3]
```diff
@@ -574,18 +581,24 @@ proc rlpxConnectIncoming*(myKeys: KeyPair, listenPort: Port, address: IpAddress,
   var handshake = newHandshake({Responder})
   handshake.host = myKeys
 
-  var authMsg: array[1024, byte]
```
[the remark this answers is not in the document]
Comment (cheatfate) [reply ethp2p/rlpx.nim]: Of course you can declare here authMsg: array[65535 + 2, byte], but is this allocation is good? maybe it is better to have authMsg = newSeq[byte](AuthMessageV4Length) which is 307 bytes, and if there is more, then allocated needed amount?

**Pull request merged**